### PR TITLE
GCW-2586 fix back link issues for request purpose

### PR DESCRIPTION
**Pull request by abhinav098**
Please review this PR.
Ticket reference: https://jira.crossroads.org.hk/browse/GCW-2582
THanks

### CODE REVIEW

File: app/controllers/request_purpose.js [hunk 1]
```diff
@@ -125,6 +125,18 @@ export default Ember.Controller.extend(cancelOrder, {
           this.transitionToRoute("order.client_information", orderId);
         }
       });
+    },
+
+    back() {
+      let prevPageName = this.get("prevPath") || this.get("previousRouteName");
+      let orderId = this.get("order.id") || this.get("orderId");
+      if (["orders.goods", "orders.booking"].indexOf(prevPageName) >= 0) {
+        this.transitionToRoute(prevPageName, orderId);
```
Comment (patrixr) [inline app/controllers/request_purpose.js]: Can't we just pop the history and go back ? Since we've been trying to move towards a normal browser behaviour

Comment (abhinav098) [reply app/controllers/request_purpose.js]: Sure, I'll try that, there was some issue with using history.back. But yeah let me try it out.

File: app/controllers/account_details.js [hunk 2]
```diff
@@ -80,7 +80,8 @@ export default Ember.Controller.extend({
     if (bookAppointment) {
       this.transitionToRoute("request_purpose", {
         queryParams: {
-          bookAppointment: true
+          bookAppointment: true,
+          prevPath: "account_details"
```
Comment (Darshan-Chauhan) [inline app/controllers/account_details.js]: Now that we're using `history.back()`, do we need this?

Comment (abhinav098) [reply app/controllers/account_details.js]: No we dont need any of the query params now. Will refactor whole routing for book appointment.

File: app/routes/static_pages.js [hunk 3]
```diff
@@ -43,10 +43,14 @@ export default Ember.Route.extend({
   actions: {
     back: function() {
       let backLink = this.get("backLink");
-      if (backLink) {
-        this.transitionTo(backLink);
+      if (!backLink) {
+        window.history.back();
+        return;
+      }
+      if (backLink.includes("request_purpose")) {
```
Comment (Darshan-Chauhan) [inline app/routes/static_pages.js]: const transitionPath = backLink.includes("request_purpose") ? 'home' : backLink;
this.transitionTo(transitionPath);

Comment (abhinav098) [reply app/routes/static_pages.js]: I've reverted this change. Wont be required now.

**Pull request merged**